snapcraft: add python-uefivars

mihalicyn opened the pull request:
We need it for "lxc config uefi <...>" extension

There is some trickiness around this thing, because for some reason snapcraft is going crazy and everything that builds after "uefivars" starts to fail with some Python errors (failed to find package XYZ).

I have workarounded it by adding "after" specifier for uefivars part.

mihalicyn commented:
I have no posted a PR for LXD yet, but I'll do that in next few days. It's safe to merge this before that as here we are just adding a new component.

MusicDin commented on snapcraft.yaml:
```diff
@@ -1375,6 +1376,16 @@ parts:
 
       patch -p1 "${CRAFT_PART_INSTALL}/snap/lxd/current/lxcfs/lxc.mount.hook" < "${CRAFT_PROJECT_DIR}/patches/lxcfs-0001-hook.patch"
 
+  uefivars:
```
I see there is only one release of python-uefivars - but should we bind it to the specific tag (`1.0.0`), or is latest ok?

tomponline replied:
we should indeed or a git sha

mihalicyn replied:
Sure, I'll do version pinning. My bad. Thanks!

tomponline replied:
@mihalicyn please can you do the version pinning and i can merge this, ta

mihalicyn replied:
Have done!

simondeziel reviewed (commented):
Agreed on the version pinning suggested by Din.

As for this new part, it should be noted that once on core24, we will be able to use the [`python3-uefivars`](https://packages.ubuntu.com/noble/python3-pyuefivars) package instead :)

tomponline commented on snapcraft.yaml:
```diff
@@ -600,6 +600,7 @@ parts:
     autotools-configure-parameters:
       - --prefix=
       - --with-json
+      - --disable-man-doc
```
This should be a separate commit to make back porting easier please

mihalicyn replied:
Have done!

mihalicyn committed:
snapcraft: disable docs build for nftables

Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

mihalicyn committed:
snapcraft: add python-uefivars

We need it for "lxc config uefi <...>" extension

There is some trickiness around this thing, because
for some reason snapcraft is going crazy and everything
that builds after "uefivars" starts to fail with some
Python errors (failed to find package XYZ).

I have workarounded it by adding "after" specifier
for uefivars part.

Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

tomponline reviewed (approved):
Thanks!
